### PR TITLE
[SPARK-10059] [yarn] Explicitly add JSP dependencies for tests.

### PR DESCRIPTION
**Pull request by vanzin**

### CODE REVIEW

File: yarn/pom.xml [hunk 1]
```diff
@@ -92,12 +92,28 @@
...
     <!-- End of shaded deps. -->
-    
+
+    <!--
+      SPARK-10059: Explicitly add JSP dependencies for tests since the MiniYARN cluster needs them.
+    -->
+    <dependency>
+      <groupId>org.eclipse.jetty.orbit</groupId>
+      <artifactId>javax.servlet.jsp</artifactId>
+      <version>2.2.0.v201112011158</version>
```
Comment (srowen) [inline yarn/pom.xml]: Does this need to be synced with the servlet API version? Ideally I suppose it does but I don't know how hard that is to plumb. I thought we had servlet 3.x in the build but you have probably looked more recently. I know JSTL is versioned differently.

Comment (vanzin) [inline yarn/pom.xml]: I started with that assumption, but these are actually the newest versions of those two artifacts.

Comment (vanzin) [inline yarn/pom.xml]: e.g. http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22javax.servlet.jsp.jstl%22

Comment (srowen) [inline yarn/pom.xml]: Oh now I remember. JSP is also versioned separately such that 2.2 is part of Servlet 3.0. Carry on. http://tomcat.apache.org/whichversion.html

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
The pyspark/yarn failure in the last run is a regression I introduced recently; I opened #8258 to fix it.

**Comment by vanzin** [comment]
YarnClusterSuite passed, so I'll merge this so that other PRs get past this point.

**Pull request closed without merging**